mitodl/mitxpro
# fix: filter enrollments instead of get in defer_enrollment

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/5848

### Description (What does it do?)
- The defer enrollment fails when there are more than one enrollments for a user in the same course runs. It's possible that the user has multiple enrollments for the same course run with different statuses.
- This PR fixes the logic in a way that we filter the data instead of getting it.
- 

### How can this be tested?
NOTE: Before testing, Make sure your edX/xPRO integration is functional

- I'd recommend producing this issue on master first. This is how you can do it
    1. Enroll in  a course run 
    2. Refund enrollment
    3. Enroll again in the same course run 
    4. Now try to defer enrollment to another run of the same course
    5. You should notice the issue
- Now, shift to this branch and test again. your defer enrollment should work fine.
